synapse/server.py

```diff
@@ -144,10 +144,10 @@
     from synapse.handlers.saml import SamlHandler
 
 
-T = TypeVar("T", bound=Callable[..., Any])
+T = TypeVar("T")
 
 
-def cache_in_self(builder: T) -> T:
+def cache_in_self(builder: Callable[..., T]) -> Callable[..., T]:
```

Comment:
I'd be tempted to replace both instances of `...` with `[object]`,  since these functions only ever take a single `self` argument. My thinking: it'd be nice if e.g. `hs.get_clock("blah blah")` would be flagged up as an error).

I'm not quite sure how that will work out though (e.g. might need to use `[]` instead of `[object]` because the self parameter is special?).

If you get the chance, I'd be interested to see if this works. But don't let this block the PR; I'd be happy to see this land as-is.

Reply:
I'll give that a try, makes sense! 👍

Reply:
I was able to make it `["HomeServer"]` 🎉